File: Data Structures & Algorithms/crawler-log-folder/submission-3.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        res = 0
        for l in logs:
            if l == './':
                continue
            if l == '../':
                if res > 0:
                    res -= 1
            else:
                res += 1
        return res

File: Data Structures & Algorithms/counting-bits/submission-3.rs
impl Solution {
    pub fn count_bits(n: i32) -> Vec<i32> {
        let mut res = vec![0; n as usize + 1];

        for i in 1..=n as usize {
            res[i] = res[i >> 1] + (i & 1) as i32;
        }

        res
    }
}
